### PR TITLE
ProxyUtils provided

### PR DESCRIPTION
**Pull request by gvolpe**
As we discussed [here](https://github.com/ruippeixotog/scala-scraper/pull/8) I created the ProxyUtils object. The README file was updated but feel free to modify it as you want. The ProxyApp example was updated using the ProxyUtils.

### CODE REVIEW

File: src/main/scala/net/ruippeixotog/scalascraper/util/ProxyUtils.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  /**
+   * Remove HTTP and HTTPS proxy configuration JVM-wide
+   */
+  def removeProxy(): Unit = setProxy(None, None)
+
+  /**
+   * Implicit conversions to String or Java null
+   */
+  private implicit def optionStrToJavaStr(value: Option[String]): String = value.getOrElse(null)
```
Comment (ruippeixotog) [inline src/main/scala/net/ruippeixotog/scalascraper/util/ProxyUtils.scala]: I think there is no need for such extra implicit conversions here, as the code for converting `Option[Int]` and `Option[String]` to `String` can be very concise inlined. For example, you can implement `setProxy` as:

```
System.setProperty("http.proxyHost", host.orNull)
System.setProperty("http.proxyPort", port.map(_.toString).orNull)
System.setProperty("https.proxyHost", host.orNull)
System.setProperty("https.proxyPort", port.map(_.toString).orNull)
```

That would remove the need for the implicits and would still provide readable code IMO.

File: src/test/scala/net/ruippeixotog/scalascraper/Examples.scala [hunk 2]
```diff
@@ -17,6 +18,14 @@ object ExampleMatchers {
   val errs = validatorsAt[Int]("error-matchers")
 }
 
+object ProxyApp extends App {
+  ProxyUtils.setProxy("localhost", 3128)
+  val browser = Browser()
+  val doc = browser.get("http://observador.pt")
+
+  println("=== OBSERVADOR HTTP & HTTPS PROXY ===")
```
Comment (ruippeixotog) [inline src/test/scala/net/ruippeixotog/scalascraper/Examples.scala]: Can you complete this example with the contents of some other scraping example, e.g. `NewsApp`? That would ensure users not only that the proxy is properly configured, but that it is really returning information.

**Comment by ruippeixotog** [comment]
Other than the comments above, LGTM! I'll probably take care of the README and Scaladoc later ;)

File: src/main/scala/net/ruippeixotog/scalascraper/util/ProxyUtils.scala (hunk 1 above)
Comment (gvolpe) [inline src/main/scala/net/ruippeixotog/scalascraper/util/ProxyUtils.scala]: Yes you are right here! In fact the code for removing the proxy it wasn't working. I should use System.clearProperty(key). Check the last commit!

File: src/test/scala/net/ruippeixotog/scalascraper/Examples.scala (hunk 2 above)
Comment (gvolpe) [inline src/test/scala/net/ruippeixotog/scalascraper/Examples.scala]: In this case I think there's not to much to add. In the last commit I added the removeProxy and a comment with the exception that you should get if you're under a proxy.

Comment (ruippeixotog) [inline src/test/scala/net/ruippeixotog/scalascraper/Examples.scala]: Ok, looks good to me then! As I do not have a proxy readily available to test, I wasn't sure that doing a GET request would be enough.

**Pull request merged**